plugins/db-mongo/src/index.ts

```diff
@@ -0,0 +1,245 @@
+import { resolve } from "path";
+import {
+  dataSource,
+  envVariables,
+  updateDockerComposeProperties,
+} from "./constants";
+import {
+  DsgContext,
+  AmplicationPlugin,
+  CreateServerDockerComposeDBParams,
+  CreateServerDockerComposeParams,
+  CreateServerDotEnvParams,
+  CreatePrismaSchemaParams,
+  Events,
+  EnumDataType,
+  EntityField,
+  Entity,
+  LookupResolvedProperties,
+  CreateSchemaFieldResult,
+  CreateServerPackageJsonParams,
+} from "@amplication/code-gen-types";
+import { ScalarType, ReferentialActions } from "prisma-schema-dsl-types";
+import * as PrismaSchemaDSL from "prisma-schema-dsl";
+import { camelCase } from "camel-case";
+import { pascalCase } from "pascal-case";
+
+class MongoPlugin implements AmplicationPlugin {
+  register(): Events {
+    return {
+      CreateServerDotEnv: {
+        before: this.beforeCreateServerDotEnv,
+      },
+      CreateServerDockerCompose: {
+        before: this.beforeCreateServerDockerCompose,
+      },
+      CreateServerDockerComposeDB: {
+        before: this.beforeCreateServerDockerComposeDB,
+        after: this.afterCreateServerDockerComposeDB,
+      },
+      CreatePrismaSchema: {
+        before: this.beforeCreatePrismaSchema,
+      },
+      CreateServerPackageJson: {
+        before: this.beforeCreateServerPackageJson,
+      },
+    };
+  }
+
+  beforeCreateServerPackageJson(
+    context: DsgContext,
+    eventParams: CreateServerPackageJsonParams
+  ) {
+    context.utils.skipDefaultBehavior = true;
+    return eventParams;
+  }
+
+  async afterCreateServerPackageJson(context: DsgContext) {
+    const staticPath = resolve(__dirname, "../static/package-json");
+    const staticsFiles = await context.utils.importStaticModules(
+      staticPath,
+      context.serverDirectories.baseDirectory
+    );
+
+    return staticsFiles;
+  }
+
+  beforeCreateServerDotEnv(
+    context: DsgContext,
+    eventParams: CreateServerDotEnvParams
+  ) {
+    eventParams.envVariables = [...eventParams.envVariables, ...envVariables];
+
+    return eventParams;
+  }
+
+  beforeCreateServerDockerCompose(
+    context: DsgContext,
+    eventParams: CreateServerDockerComposeParams
+  ) {
+    eventParams.updateProperties.push(...updateDockerComposeProperties);
+    return eventParams;
+  }
+
+  beforeCreateServerDockerComposeDB(
+    context: DsgContext,
+    eventParams: CreateServerDockerComposeDBParams
+  ) {
+    context.utils.skipDefaultBehavior = true;
+    return eventParams;
+  }
+
+  async afterCreateServerDockerComposeDB(context: DsgContext) {
+    const staticPath = resolve(__dirname, "../static");
```

Comment:
now, you need to put the `docker-compose.db.yml` also under a folder and change this line to:
```suggestion
    const staticPath = resolve(__dirname, "../static/docker-compose");
```